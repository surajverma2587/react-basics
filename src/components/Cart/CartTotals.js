import React from 'react';
import PropTypes from 'prop-types';

const CartTotals = (/* props object */) => {
  /* bind the data from the props object to the html template */
  /* validate the props and render the component IF valid ELSE null */
  /* replace the bottom return statement with the populated html template */
  return (
    <div>
      CartTotals Component
    </div>
  );
};

/* display name for the component */
CartTotals.displayName = 'CartTotals';

CartTotals.propTypes = {
  /* add Proptypes here */
};

export default CartTotals;