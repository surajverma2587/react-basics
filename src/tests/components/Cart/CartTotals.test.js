import React from 'react';
import { shallow } from 'enzyme';

import CartTotals from '../../../components/Cart/CartTotals';

const props = {
  subTotal: 10,
  taxes: 2,
  delivery: 3,
  total: 15,
};

describe('CartTotals Component', () => {
  describe('when props are valid', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<CartTotals {...props} />);
    });
    
    it('should render snapshot of the component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render the component', () => {
      const cartTotals = wrapper.find('.cart_totals');
      expect(cartTotals.exists()).toBe(true);
    });
  
    it('should render the sub total', () => {
      const subTotal = wrapper.find('.cart_totals--price').at(0);
      expect(subTotal.text()).toEqual(`£${props.subTotal}`);
    });

    it('should render the taxes', () => {
      const taxes = wrapper.find('.cart_totals--price').at(1);
      expect(taxes.text()).toEqual(`£${props.taxes}`);
    });

    it('should render the delivery', () => {
      const delivery = wrapper.find('.cart_totals--price').at(2);
      expect(delivery.text()).toEqual(`£${props.delivery}`);
    });

    it('should render the total', () => {
      const total = wrapper.find('.cart_totals--price').at(3);
      expect(total.text()).toEqual(`£${props.total}`);
    });
  });
  
  describe('when props are invalid', () => {
    it('should not render the component when props are not present', () => {
      const wrapper = shallow(<CartTotals />);
      expect(wrapper.find('.cart_totals').exists()).toBe(false);
    });
  
    it('should not render the component when any prop is invalid', () => {
      const invalidProps = [
        { ...props, subTotal: null },
        { ...props, taxes: null },
        { ...props, delivery: null },
        { ...props, total: null },
      ];
  
      invalidProps.map(invalidProp => {
        const wrapper = shallow(<CartTotals {...invalidProp} />);
        return expect(wrapper.find('.cart_totals').exists()).toBe(false);
      });
    });
  });
});
