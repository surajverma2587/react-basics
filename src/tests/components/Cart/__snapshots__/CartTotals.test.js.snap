// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotals Component when props are valid should render snapshot of the component 1`] = `
<div
  className="cart_totals"
>
  <div
    className="row"
  >
    <div
      className="col-sm-6 cart_totals--info"
    >
      <span>
        Sub-Total:
      </span>
    </div>
    <div
      className="col-sm-6 cart_totals--price"
    >
      <span>
        £
        10
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6 cart_totals--info"
    >
      <span>
        Taxes:
      </span>
    </div>
    <div
      className="col-sm-6 cart_totals--price"
    >
      <span>
        £
        2
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6 cart_totals--info"
    >
      <span>
        Delivery:
      </span>
    </div>
    <div
      className="col-sm-6 cart_totals--price"
    >
      <span>
        £
        3
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6 cart_totals--info"
    >
      <span>
        TOTAL:
      </span>
    </div>
    <div
      className="col-sm-6 cart_totals--price"
    >
      <span>
        £
        15
      </span>
    </div>
  </div>
</div>
`;
